x-pack/legacy/plugins/siem/public/components/ml/tables/convert_anomalies_to_network.ts

```diff
@@ -25,3 +31,15 @@ export const convertAnomaliesToNetwork = (anomalies: Anomalies | null): Anomalie
     }, []);
   }
 };
+
+export const getNetworkFromEntity = (anomaly: Anomaly, ip?: string): boolean => {
+  if (anomaly.entityName === 'source.ip' || anomaly.entityName === 'destination.ip') {
```

Comment:
nit - to simplify/readable
```
if (!['source.ip', 'destination.ip'].contains(anomaly.entityName)) {
  return false;
}
if (ip == null) {
  return true;
}
return anomaly.entityValue === ip;
```

Reply:
For this, I took that above code but put it within `types.ts` and added this to the `types.ts`:

```ts
const sourceOrDestination = ['source.ip', 'destination.ip'];

export const isDestinationOrSource = (value: string | null): value is DestinationOrSource =>
  value != null && sourceOrDestination.includes(value);
```

Notice the `is DestinationOrSource` and the `is` keyword. That makes it so typescript doesn't complain as much whenever I pass in something could be null or something that is a string.

I then reuse this within other parts of the code.